source/robot_model/test/tests/test_model.cpp

```diff
@@ -56,43 +56,35 @@ TEST_F(RobotModelTest, TestNumberOfJoints) {
   EXPECT_EQ(franka->get_number_of_joints(), 7);
 }
 
-TEST_F(RobotModelTest, TestForwardGeometryJointStateSize) {
-  state_representation::JointState dummy = state_representation::JointState(robot_name, 6);
-  EXPECT_THROW(franka->forward_geometry(dummy), exceptions::InvalidJointStateSizeException);
-}
-
-TEST_F(RobotModelTest, TestForwardGeometry) {
-  EXPECT_EQ(franka->forward_geometry(joint_state).get_position(),
-            franka->forward_geometry(joint_state, "panda_link8").get_position());
-}
-
-TEST_F(RobotModelTest, TestForwardGeometryInvalidFrameName) {
-  EXPECT_THROW(franka->forward_geometry(joint_state, "panda_link99"), exceptions::FrameNotFoundException);
-}
 
 TEST_F(RobotModelTest, TestJacobianJointNames) {
   state_representation::JointState dummy = state_representation::JointState(robot_name, 7);
   state_representation::Jacobian jac = franka->compute_jacobian(dummy);
   for (int i = 0; i < 7; ++i) {
     std::string jname = "panda_joint" + std::to_string(i + 1);
-    EXPECT_TRUE(jname.compare(jac.get_joint_names()[i]) == 0);
+    EXPECT_STREQ(jname.c_str(), jac.get_joint_names()[i].c_str());
```

Comment:
I believe `EXPECT_EQ` is sufficient instead of casting to `char*`

Reply:
I looked this up and you're right, I have been using it wrong myself. 

I thought the specific `STREQ` macro would have better behaviour in the test result formatting or something, but it's really just a specific implementation for C style strings. `EXPECT_EQ` checks for pointer equality of objects, so using that on a C style string (`char *`) will be comparing the memory location instead of the content. C++ strings are objects though, so `_EQ` comparison works on the object content properly.

Therefore, using `EXPECT_STREQ(somestring.c_str()...` is actually a workaround _around_ a workaround, and `EXPECT_EQ` is much more preferable for C++ string objects.